lua/lspconfig/server_configurations/drools_lsp.lua

```diff
@@ -1,35 +1,60 @@
 local util = require 'lspconfig.util'
 
-local get = function(config, section, key)
-  return vim.tbl_get(config.settings, section, key) and config.settings[section][key] or nil
+local function get(config, segments)
+  if vim.tbl_get(config, unpack(segments)) then
+    local cfg = config
+    for _, seg in ipairs(segments) do
+      cfg = cfg[seg]
+    end
+    return cfg
+  end
+  return nil
 end
 
-local get_java_bin = function()
-  local java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
-  if vim.fn.has 'win32' == 1 then
-    java_bin = java_bin .. '.exe'
+local function get_java_bin(config)
+  local java_bin = get(config, { 'drools', 'java', 'bin' })
+  if java_bin == nil then
+    java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
+    if vim.fn.has 'win32' == 1 then
+      java_bin = java_bin .. '.exe'
+    end
   end
   return java_bin
 end
 
+local function get_java_opts(config)
```

Comment:
then this no need

Reply:
That function exists because if `{ 'drools', 'java', 'opts' }` is not configured by the user, then I want to return an empty table instead of nil.